src/util/bit_util.cuh

```diff
@@ -0,0 +1,96 @@
+
+/*
+ * Copyright 2018 BlazingDB, Inc.
+ *     Copyright 2018 Alexander Ocsa <alexander@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+namespace gdf {
+namespace util {
+
+static constexpr int ValidSize = 32;
+using ValidType = uint32_t;
+
+__host__ __device__ __forceinline__
+  size_t
+  valid_size(size_t column_length)
```

Comment:
So I'm confused about this function. Is this supposed to return the number of `gdf_valid_type`s (i.e. `unsigned char`) it takes to give you enough bits for `column_length` rows?

If so, I don't think this logic makes sense because you're using `ValidSize = 32`, but shouldn't it be `ValidSize = sizeof(gdf_valid_type) * 8'?

I think in general, try to avoid hardcoding any of the information about how large the `gdf_valid_type` type is and make it all a function of `sizeof(gdf_valid_type)`, because this morning someone was already asking about using 32bit variables instead of 8 bit variables for the `gdf_valid_type`.

Reply:
@jrhemstad I agree so the exact number of bytes for gdf_column.valid is (column_size + 8 - 1) / 8; where comun_size is gdf_column.size. However I implemented some kernel to works with valid data with a size size that is multiple of 32 because I optimize it in order to match the warp_size.

Reply:
Ah, I see, that makes more sense.